File: SWEA/Java/[D2] 1859 백만 장자 프로젝트/백만장자프로젝트.java
import java.util.Scanner;

public class 백만장자프로젝트 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		// 테스트케이스 개수
		int T = sc.nextInt();

		// T개의 테스트케이스
		for (int t = 1; t <= T; t++) {
			// 연속된 날짜 수
			int N = sc.nextInt();

			// 각 날의 매매가
			int[] price = new int[N];

			for (int i = 0; i < N; i++) {
				price[i] = sc.nextInt();
			}

			// 이익
			long plus = 0;

			// 현재 최대 매매가
			int max = price[N - 1];

			// 뒤에서부터 값을 비교
			// 현재 최대 매매가보다 값이 작으면 지출, 값이 크거나 같으면 지금까지의 이익을 구하고 max값 변경
			for (int i = N - 2; i >= 0; i--) {
				// 현재 최대 매매가보다 작을 때
				if (price[i] < max) {
					plus = plus - price[i] + max;
				}

				// 현재 최대 매매가보다 크거나 같을 때
				else {
					max = price[i];
				}
			}

			// 출력
			System.out.printf("#%d %d\n", t, plus);
		}
	}
}

File: SWEA/Python/[D2] 1859 백만 장자 프로젝트/백만 장자 프로젝트.py
# 테스트 케이스 개수
T = int(input())
for t in range(1, T + 1):
    # 일 수
    N = int(input())

    # N일 동안의 매매가
    sales = list(map(int, input().split()))

    # 최대 이익
    plus = 0

    # 상한가
    mx = sales[-1]
    for i in range(N - 2, -1, -1):
        # 오늘 매매가
        price = sales[i]

        # 상한가보다 작다면 이익
        if price < mx:
            plus += mx - price

        # 상한가보다 크다면 상한가 바꾸기
        elif price > mx:
            mx = price

    print(f'#{t} {plus}')